cms/cms_config.py

```diff
@@ -0,0 +1,38 @@
+from django.core.exceptions import ImproperlyConfigured
+
+from cms.app_base import CMSAppExtension, CMSAppConfig
+from cms.cms_wizards import cms_page_wizard, cms_subpage_wizard
+from cms.wizards.wizard_base import Wizard
+
+
+class CMSCoreConfig(CMSAppConfig):
+    cms_enabled = True
+    cms_wizards = [cms_page_wizard, cms_subpage_wizard]
+
+
+class CMSCoreExtensions(CMSAppExtension):
+
+    def __init__(self):
+        self.wizards = {}
+
+    def configure_wizards(self, cms_config):
+        """
+        Adds all registered wizards from apps that define them to the
+        wizards dictionary on this class
+        """
+        if not hasattr(cms_config, 'cms_wizards'):
+            # The cms_wizards settings is optional. If it's not here
+            # just move on.
+            return
+        if type(cms_config.cms_wizards) != list:
+            raise ImproperlyConfigured("cms_wizards must be a list")
+        for wizard in cms_config.cms_wizards:
+            if not isinstance(wizard, Wizard):
+                raise ImproperlyConfigured(
+                    "all wizards defined in cms_wizards must inherit "
+                    "from cms.wizards.wizard_base.Wizard")
+            else:
```

Comment:
So I did this locally and am now struggling with random tests failing. What seems to be happening is:

1. The test setup is loading cms and sampleapp as expected and their wizards are getting registered
1. The individual tests use the `override_settings` decorator with INSTALLED_APPS and include the cms app on the revised list, which in turn triggers another autodiscover thing because the cms's ready method is getting called twice. And therefore raises an error.

I can think of some ways of fixing this, but all these ways seem really hacky and will be really annoying and unobvious for future test writers. Do you have an idea for a nice way of doing this?
Alternatively, because we're using a dict, nothing bad is actually going to happen if a wizard is added to the dict twice (it'll just override the previous wizard), so we could just change this to a `logger.warning`? Really our code can cope with double registration of wizards, it's just that the developer has done something fishy if that's what's happening, so we should warn them.

Reply:
Is the app config instance setup on the first load, persisting through to the second load on the tests that use `override_settings`?

Reply:
@czpython I believe so, yes. I seem to be struggling on a similar issue with the integration test on backwards compatibility (needed for the work on retaining the autodiscover of cms_wizards) as well. I've just pushed that.